fe/fe-core/src/main/java/org/apache/doris/qe/PointQueryExecutor.java

```diff
@@ -208,21 +409,61 @@ public RowBatch getNext() throws Exception {
         if (candidateBackends == null || candidateBackends.isEmpty()) {
             return new RowBatch();
         }
+        OlapTable olapTable = shortCircuitQueryContext.scanNode.getOlapTable();
+        List<Column> keyColumns = olapTable.getBaseSchemaKeyColumns();
+        for (int i = 0; i < keyColumns.size(); ++i) {
+            Column column = keyColumns.get(i);
+            if (olapTable.isDistributionColumn(column.getName())) {
```

Comment:
the logic is wrong, what if column is both distribution column and partitionKeyColumns

Reply:
[Here](https://github.com/apache/doris/pull/39468/files?diff=split&w=0#:~:text=if%20(olapTable.getPartitionColumns().contains(column))%20%7B) after distribution prunnig, we will use the partition information to check whether the column is a partition column, so as to prune as many buckets as possible.